### ISSUE TITLE
Loosen domain name scrubbing in `span.domain` tag

### ISSUE THREAD

**Issue by gggritso** [getsentry/relay#3573]
### Problem Statement

Relay scrubs domain addresses like `my.app.service.com` to something like `*.service.com`. This is okay for a lot of cases, but for _some_ users, subdomains are meaningful to more than just 2 segments. This scrubbing causes different domains to be mashed together in the Requests view.

### Solution Brainstorm

- allow more segments (e.g., leave the last _three_ segments, or more)
- no scrubbing at all (surely we can't allow that, there are too many unique values)
- customizable domain scrubbing (let people to configure how domains are scrubbed)
- allow self-hosted users to turn domain scrubbing off (people who have their own infra might want to turn this off, and bear the weight of high cardinality)

**Comment by jjbayer** [comment]
Domain scrubbing is definitely arbitrary right now, because the second level domain might be just as high-cardinality as the subdomain.

@gggritso what if we approach this in a data-driven way, i.e. collect a sample of unscrubbed domains from from the spans dataset and see what kind of scrubbing would yield the best results? One thing we could immediately evaluate is subdomain removal vs. scrubbing of integer / hex IDs. Is this something you'd be willing to drive?

**Comment by aldy505** [comment]
I think I'd prefer this one:

> allow self-hosted users to turn domain scrubbing off (people who have their own infra might want to turn this off, and bear the weight of high cardinality)

It doesn't disturbs the behavior on SaaS and it has a toggle for it.

**Comment by gggritso** [comment]
@jjbayer that approach makes sense! I'd be happy to do that at some point, though not sure when I'll have a chance

**Comment by aldy505** [comment]
Since the implementation on https://github.com/getsentry/relay/pull/3813 also works for this issue, we should close it.

**Comment by cobyeastwood183** [comment]
Have a customer writing in who would like the option to either loosen or disable the scrubbing of domains on Sentry SaaS (gets displayed on frontend insights page). Their need is very similar to this issues problem statement, they want to see second and third level domains. Reopening this to see if thats something we could do?

**Comment by cobyeastwood183** [comment]
https://sentry.zendesk.com/agent/tickets/161902

**Comment by loewenheim** [comment]
We'll discuss it in the team.

**Comment by Dav1dde** [comment]
@gggritso we did the domain thing for metrics cardinality, with metrics gone should we just remove it alltogether?

**Comment by gggritso** [comment]
@Dav1dde not yet! The domain thing is also for user experience, if we remove the wildcards a lot of domains will get unrolled into high cardinality strings, and then domain list will be a lot less useful because of the lack of grouping

**Comment by Dav1dde** [comment]
Talked to @gggritso, we realized there is no easy way forward here without keeping away breaking changes and longterm George has a solution in mind that will also solve this but still requires more fundamental work to be done first.

tldr; it's being worked on but may take a while

**Comment by Dav1dde** [comment]
This should no longer be a concern with upcoming span streaming SDKs.
